packages/valory/connections/ledger/connection.py

```diff
@@ -203,4 +180,6 @@ def _handle_done_task(self, task: asyncio.Future) -> Optional[Envelope]:
                 message=response_message,
                 context=request.context,
             )
-        return response_envelope
+
+        # not handling `asyncio.QueueFull` exception, because the maxsize we defined for the Queue is infinite
+        self.response_envelopes.put_nowait(response_envelope)
```

Comment:
are you going to address the full queue issue as well? Leaving it at infinite size seems like a bad idea, not?

Reply:
That would introduce an arbitrary magic value defining the queue size. It's not a bad question, but the answer is more involved as there are other areas of the framework where queues can build up.